src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergDataTypeTest.java

```diff
@@ -0,0 +1,441 @@
+package net.snowflake.ingest.streaming.internal.datatypes;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZoneOffset;
+import java.util.Map;
+import java.util.UUID;
+import net.snowflake.ingest.TestUtils;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestChannel;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+@Ignore("This test can be enabled after server side Iceberg EP support is released")
+public class IcebergDataTypeTest extends AbstractDataTypeTest {
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Before
+  public void before() throws Exception {
+    super.before(true);
+  }
+
+  @Test
+  public void testBoolean() throws Exception {
+    testIcebergIngestion("boolean", true, new BooleanProvider());
+    testIcebergIngestion("boolean", false, new BooleanProvider());
+    testIcebergIngestion("boolean", 1, true, new BooleanProvider());
+    testIcebergIngestion("boolean", "false", false, new BooleanProvider());
+
+    SFException ex =
+        Assert.assertThrows(
```

Comment:
pl use fluent assertions (assertj) as they (a) have test code that reads like a story, and (b) generate very well formatted error messages. Assert.assert*() has the most useless error messages when things fail unless every single call to the method has an associated string message that details the current state of whatever was being asserted on.

For instance:
assertThat(ex).isNotNull();
assertThat(ex.getVendorCode).isEqualTo(ErrorCode.INVALID_FORMAT_ROW.getMessageCode());
assertThatCode(() -> lambda).doesNotThrow();
assertThatThrownBy(() -> lambda).hasCause();

For this you'll have to add the assertj library (I added it once and it failed in one of the mergegates, did not have the time to find and fix what happened!). Feel free to remove the org.hamcrest import from this project.

Reply:
Switched to assertj library.